### PR TITLE
 zos: clear poll events on every iteration

### PR DESCRIPTION
**Pull request by jBarz**
Also includes a number of other minor fixes which can be squashed.

### CODE REVIEW

File: src/unix/os390-syscalls.c [hunk 1]
```diff
@@ -141,7 +141,7 @@ static void init_message_queue(uv__os390_epoll* lst) {
   } msg;
 
   /* initialize message queue */
-  lst->msg_queue = msgget(IPC_PRIVATE, 0622 | IPC_CREAT);
+  lst->msg_queue = msgget(IPC_PRIVATE, 0600 | IPC_CREAT);
```
Comment (bnoordhuis) [inline src/unix/os390-syscalls.c]: This should probably stay a separate commit.

File: src/unix/os390-syscalls.c [hunk 2]
```diff
@@ -294,10 +297,16 @@ int epoll_wait(uv__os390_epoll* lst, struct epoll_event* events,
 
     ev.fd = pfds[i].fd;
     ev.events = pfds[i].revents;
-    events[reventcount++] = ev;
+    if (pfds[i].revents & POLLIN && pfds[i].revents & POLLOUT)
+      reventcount += 2;
+    else if (pfds[i].revents & (POLLIN | POLLOUT))
+      ++reventcount;
```
Comment (bnoordhuis) [inline src/unix/os390-syscalls.c]: Maybe this?
```c
reventcount += !!(pfds[i].revents & POLLIN);
reventcount += !!(pfds[i].revents & POLLOUT);
```
And since you're referencing `pfds[i]` a lot I'd maybe assign it to a `struct pollfd *pfd`.

File: src/unix/os390-syscalls.c [hunk 3]
```diff
@@ -493,7 +502,7 @@ ssize_t os390_readlink(const char* path, char* buf, size_t len) {
 
 
 size_t strnlen(const char* str, size_t maxlen) {
-  void* p = memchr(str, 0, maxlen);
+  char* p = memchr(str, 0, maxlen);
```
Comment (bnoordhuis) [inline src/unix/os390-syscalls.c]: This should be its own commit.

**Comment by jBarz** [comment]
Just curious about what is good practice.
Is 2 approvals good enough to push a commit? Or should I wait for another one?

**Comment by bnoordhuis** [comment]
In principle one sign-off is enough.  In practice you might want to give it another day or two to give other maintainers time to chime in, unless it's trivial or arcane stuff (say, a typo fix or an AIX bug fix that no one outside IBM cares about.)

Just use your good judgment and if you land stuff too soon, one of us will tell you. :-)

**Comment by jBarz** [comment]
Merged in https://github.com/libuv/libuv/commit/71bb0cc25a5286e319d092dbd62837e0baa04a58
